src/components/DataViewFilterText/DataViewFilterText.tsx

```diff
@@ -119,8 +119,13 @@ function DataViewFilterText(props: DataViewFilterTextProps) {
 	// 	throwOnInvalid : true
 	// });
 
+	if (props.comparisonDefault && validComparisons.find( validComparison => validComparison.value === props.comparisonDefault) === undefined)
+		throw new Error("The selected comparison is not a valid comparison")
+
 	const [anchorEl, setAnchorEl] = useState(null);
-	const comparison = props.data.comparison || (props.comparisonDefault || "equals");
+	const comparison = props.args && props.args.comparisons
+		? props.data.comparison || (props.comparisonDefault || "equals")
+		: props.data.comparison || "equals";

```

Comment:
I don't know where to put this but I was poking around trying to see if it worked for all components using DataViewFilterText... it does but only if the filter is optional (see DataViewKitchenSink), for primary filters it doesn't seem to work. I'm guessing this has something to do with some useEffects or memoization or something like that but let's talk about it tomorrow.